src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -15294,28 +15294,113 @@ def dt_normalize(self) -> None:
             "Snowpark pandas doesn't yet support the method 'Series.dt.normalize'"
         )
 
-    def dt_month_name(self, locale: Optional[str] = None) -> None:
+    def dt_month_name(self, locale: Optional[str] = None) -> "SnowflakeQueryCompiler":
         """
         Args:
             locale: Locale determining the language in which to return the month name.
 
         Returns:
             New QueryCompiler containing month name.
         """
-        ErrorMessage.not_implemented(
-            "Snowpark pandas doesn't yet support the method 'Series.dt.month_name'"
+        if locale is not None:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas 'Series.dt.month_name' method doesn't yet support 'locale' parameter"
+            )
+        internal_frame = self._modin_frame
+        snowpark_column = builtin("decode")(
```

Comment:
why not use https://docs.snowflake.com/en/sql-reference/functions/monthname?

Reply:
also https://docs.snowflake.com/en/sql-reference/functions/dayname

Reply:
That's because they are spelled differently in pandas vs Snowflake. Full names in pandas vs abbreviated names in Snowflake - both for days and months.